### ISSUE TITLE
Cannot load plugins via command line

### ISSUE THREAD

**Issue by tnunamak** [eslint/eslint#3383]
Going by the [plugin option docs](http://eslint.org/docs/user-guide/command-line-interface#plugin), any plugins I specify with the `--plugin` option are ignored. On the other hand, if I specify a plugins property in `.eslintrc`, they are loaded.

Digging into the `lib/cli-engine` and `lib/config` modules, it seems that `config.plugins` is undefined [here](https://github.com/eslint/eslint/blob/master/lib/cli-engine.js#L181):

```javascript
    config = configHelper.getConfig(filePath);
    loadPlugins(config.plugins);
```

Using v1.1.0

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by nzakas** [comment]
Just to be clear, you are reporting that `--plugin` doesn't work at all?

**Comment by IanVS** [comment]
I can confirm, it is not working at all for me either.  It doesn't error out, but specifying a plugin on the CLI seems to do nothing.

**Comment by gyandeeps** [comment]
To add, I dint see any unit test in `cli` file. That is shocking ? :worried:

**Comment by IanVS** [comment]
Working on it.

**Comment by IanVS** [comment]
@gyandeeps it's a tough one to test.  The plugin has to be installed into node_modules, and the CLI option takes only the name.  The only way I've come up with so far is to create a fixture plugin and include it in `devDependencies`.  Or, is there a better way to use a fixture plugin?

**Comment by gyandeeps** [comment]
You can use `proxyquire` to fake certain modules. For examples please look at unit tests inside `config` file. Its very straight forward that ways.
